app/assets/javascripts/test/backend-snapshot-tests/teamstructure.e2e.js

```diff
@@ -0,0 +1,170 @@
+/* eslint import/no-extraneous-dependencies: ["error", {"peerDependencies": true}] */
+/* eslint-disable import/first */
+// @flow
+import {
+  tokenUserDefault,
+  tokenUserAAB,
+  tokenUserAAC,
+  tokenUserABA,
+  tokenUserBAA,
+  setCurrToken,
+} from "../enzyme/e2e-setup";
+import test from "ava";
+import _ from "lodash";
+import * as api from "admin/admin_rest_api";
+
+/*
+TEAM STRUCTURE USED FOR TESTING:
+
+Organizations:
+Connectomics department (Organization A), OrganizationB
+
+Teams:
+Connectomics department (Team AA), Team AB, Team BA, test1, test2
+(the teams test1 and test2 existed before)
+
+User:
+default (UserAAA), User AAB, User AAC, User ABA, User BAA
+
+++----------------------------------------------------------------------------------------------------------------------++--------------------------------------++
+||	Connectomics department (Organization A)																			                                      ||	Organization B 						          ||
+||  Admin: scmboy																										                                                    ||  Admin: userBAA					          	||
+++-----------------------------------+--------------------------+---------------------------+---------------------------++--------------------------------------++
+|| Connectomics department (Team AA) | 		Team AB				        | 		test1				          | 		test2				          ||  TeamBA 								              ||
+++-----------------------------------+--------------------------+---------------------------+---------------------------++--------------------------------------++
+|| scmboy (userAAA)	(teamMng)		     | 		userABA (teamMng.)    | 		scmboy            	  | 		scmboy (teamMng.)	    ||  userBAA	(teamMng.)			         		||
+|| userAAB (teamMng.)		   		       | 						            	| 					              	| 						            	||  							                  		||
+|| userAAC					   		           | 						            	| 					            		| 						            	||  							                  		||
+++-----------------------------------+--------------------------+---------------------------+---------------------------++--------------------------------------++
+ */
+
+// Teams
+test("teams_userDefault", async t => {
+  await setCurrToken(tokenUserDefault);
+  const teams = _.sortBy(await api.getTeams(), team => team.name);
+  t.snapshot(teams[0], { id: "teams_userDefault" });
+});
+
+test("teams_userABA", async t => {
+  await setCurrToken(tokenUserABA);
+  const teams = _.sortBy(await api.getTeams(), team => team.name);
+  t.snapshot(teams[0], { id: "teamsABA" });
+});
+
+test("teams_userBAA", async t => {
+  await setCurrToken(tokenUserBAA);
+  const teams = _.sortBy(await api.getTeams(), team => team.name);
+  t.snapshot(teams[0], { id: "teams_userBAA" });
+});
+
+test("teams_delete_userABA", async t => {
+  // the teamManager is not allowed to delete the team
+  await setCurrToken(tokenUserABA);
+  try {
+    await api.deleteTeam("69882b370d889b84020efd4f");
```

Comment:
What about the following?

```js
	const error = await t.throws(async () => {
		await api.deleteTeam("69882b370d889b84020efd4f");
	}, Error);
    t.is(error.message, "not authorized...");
```

This tells ava to expect an exception, when triggering deleteTeam (see [here](https://github.com/avajs/ava#throwsthrower-expected-message)). Putting an assertion on the error afterwards ensures that `deleteTeam` fails for the right reason. For example, some back-end might adapt the deleteTeam API so that another parameter is needed. Then, this test would still succeed. If another back-end change breaks the validation of the user token, this test won't catch this bug since the api call fails as expected (but due to another reason).